Character.cpp

```diff
@@ -98,20 +98,37 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        getCharacterReward(*initialHitPoints, hitPoints);
+        getCharacterReward(*initialArmorLevel, armor);
+        getCharacterReward(*initialAttackDamage, attackDamage);
+    
+        
+        //assert(false);
         std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
     }
 }
 
+
+void Character::getCharacterReward(int& initial, int& current)
+{
+    if( current < initial )
+    {
+        current = initial + (initial * 0.1);
+    }
+    
+}
```

Comment:
This function needs to implement all 3 requirements:
```
        When you defeat another Character: 
            a) your stats are restored to their initial value if they are lower than it.
            b) your stats are boosted 10%
            c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
```
Do not combine `a, b, and c`.
you are currently combining them into a single expression.
- [ ] revise